r0bynh00d/react-beginner-demo
# Step 2: Initialise the Test Environment

#### What does this PR do?
This PR implements the CI/CD environment to run tests, build and deploy the app...with README badges showing the build status and other useful info about the test environment. 

We use the following toolbox: 
- [Jest](https://jestjs.io)
- [Enzyme](https://enzymejs.github.io/enzyme)
- [Travis CI](https://travis-ci.com)
- [Code Climate](https://codeclimate.com)
- [CircleCI](https://circleci.com) ~ optional

#### Description of Task to be completed?
- sync the GitHub repo at Travis-CI.com.
- setup Travis CI configuration `.travis.yml` with README badge.
- setup minimal test environment based on Jest/Enzyme.
- setup Code Climate configuration `.codeclimate.yml` with README badges (for test coverage and mainainability).
- [BONUS] setup CircleCI configuration `.circleci/config.yml` with README badge.

#### How should this be manually tested?
- Push some commits to the project's GitHub repo.
- Open a Pull Request on GitHub for the new commits.
- View the status of the Continuous Integration checks.
- Also, check the project README on GitHub to see status badges from the different CI/CD tools.

#### Any background context you want to provide?
**TDD** - This is required to automatically run tests and other preliminary build steps required to ship the next iteration of the product.
**Jest** - A popular testing library for JavaScript used by Facebook.
**Enzyme** - A JavaScript testing utility made, by AirBnB, specifically for React.

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots

##### Travis CI (GitHub integration) checks in Pull Request
![Travis CI setup](https://user-images.githubusercontent.com/65714243/83361456-63e4af80-a389-11ea-8a61-62bb15c04927.png)
